tests/transforms/test_experimental/test_transform_dispatcher.py

```diff
@@ -0,0 +1,344 @@
+
+
+def no_callable_return(
+    tape: qml.tape.QuantumTape,
+) -> (Sequence[qml.tape.QuantumTape], qml.tape.QuantumTape):
+    """Transform without callable return."""
+    return tape, lambda x: x
+
+
+non_valid_transforms = [
+    non_callable,
+    no_processing_fn_transform,
+    no_tape_sequence_transform,
+    no_tape_transform,
+    no_quantum_tape_transform,
+    no_callable_return,
+]
+
+
+##########################################
+# Valid transforms
+
+
+def first_valid_transform(
+    tape: qml.tape.QuantumTape, index: int
+) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """A valid transform."""
+    tape.circuit.pop(index)
+    return [tape], lambda x: x
+
+
+def second_valid_transform(
+    tape: qml.tape.QuantumTape, index: int
+) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """A valid trasnform."""
+    tape1 = tape.copy()
+    tape2 = tape.circuit.pop(index)
+
+    def fn(results):
+        return qml.math.sum(results)
+
+    return [tape1, tape2], fn
+
+
+valid_transforms = [first_valid_transform, second_valid_transform]
+
+
+##########################################
+# Non-valid expand transform
+def multiple_args_expand_transform(
+    tape: qml.tape.QuantumTape, index: int
+) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """Multiple args expand fn."""
+    tape.circuit.pop(index)
+    return [tape], lambda x: x
+
+
+# Valid expand transform
+def expand_transform(tape: qml.tape.QuantumTape) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """A valid expand transform."""
+    return [tape], lambda x: x
+
+
+class TestTransformDispatcher:
+    """Test the transform function (validate and dispatch)."""
+
+    @pytest.mark.parametrize("valid_transform", valid_transforms)
+    def test_integration_dispatcher_with_valid_transform(self, valid_transform):
+        """Test that no error is raised with the transform function and that the transform dispatcher returns
+        the right object."""
+
+        dispatched_transform = transform(valid_transform)
+
+        # Applied on a tape
+        tapes, fn = dispatched_transform(tape_circuit, 0)
+
+        assert isinstance(tapes, Sequence)
+        assert callable(fn)
+
+        # Applied on a qfunc (return a qfunc)
+        qfunc = dispatched_transform(qfunc_circuit, 0)
+        assert callable(qfunc)
+
+        # Applied on a qnode (return a qnode with populated the program)
+        @qml.qnode(device=dev)
+        def qnode_circuit(a):
+            """QNode circuit."""
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PauliX(wires=0)
+            qml.RZ(a, wires=1)
+            return qml.expval(qml.PauliZ(wires=0))
+
+        qnode = dispatched_transform(qnode_circuit, 0)
+        assert isinstance(qnode, qml.QNode)
+        assert isinstance(qnode.transform_program, list)
+        assert isinstance(qnode.transform_program[0], qml.transforms.core.TransformContainer)
+
+    def test_queuing_qfunc_transform(self):
```

Comment:
Would it be useful to test that the resulting qfunc and tape have had the correct effect by the transform? For example, if we define a transform to remove an operation from the tape, would it help assessing that such operation has been indeed removed?

Reply:
Yes I will add it!